File: sprint03/KDH/PGS_12977.py
# 소수 만들기
from itertools import combinations


def solution(nums):
    list_ = list(combinations(nums, 3))

    def is_decimal(num):
        cnt = 0
        for i in range(1, num):
            if num % i == 0:
                cnt += 1

        if cnt == 1:
            return True
        else:
            return False

    answer = 0

    for a, b, c in (list_):
        if is_decimal(a+b+c):
            answer += 1

    return answer

File: sprint03/JMS/PGS_12977.cpp
#include <vector>
#include <iostream>
using namespace std;

bool IsPrime(int num){
    if(num==1)
        return true;
    
    for(int i=2;i<num;i++){
        if(num%i == 0)
            return false;
    }
    
    return true;
}


int solution(vector<int> nums) {
    int answer = 0;
    
    for(int i=0;i<nums.size();i++){
        for(int j=i+1;j<nums.size();j++){
            for(int k =j+1;k<nums.size();k++){
                if(IsPrime(nums[i]+nums[j]+nums[k]))
                    answer++;
            }
        }
    }
    
    return answer;
}
